#951 broke building for x86_64-pc-windows-gnu and i686-pc-windows-gnu

Output at the end:
```
C:/PROGRA~1/MINGW-~1/X86_64~1.0-P/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lvcruntime
collect2.exe: error: ld returned 1 exit status
mingw32-make.exe[2]: *** [CMakeFiles\SDL2.dir\build.make:2365: SDL2.dll] Error 1
mingw32-make.exe[1]: *** [CMakeFiles\Makefile2:105: CMakeFiles/SDL2.dir/all] Error 2
mingw32-make.exe: *** [Makefile:129: all] Error 2
thread 'main' panicked at '
command did not execute successfully, got: exit code: 2

build script failed, must exit now', C:\Users\jrb0001\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.42\src\lib.rs:861:5
```

It looks like the patching infrastructure needs to be extended for target specific patches or is there a better way?

--- a/sdl2-sys/patches/SDL2-2.0.9-CMakeLists.txt.patch
+++ b/sdl2-sys/patches/SDL2-2.0.9-CMakeLists.txt.patch
@@ -1,11 +1,13 @@
 --- CMakeLists.txt	2018-10-31 08:07:22.000000000 -0700
 +++ CMakeLists.txt	2019-12-04 21:50:07.606700200 -0800
-@@ -1292,7 +1292,7 @@
-   endif()
- 
+@@ -1294,6 +1294,10 @@
    # Libraries for Win32 native and MinGW
--  list(APPEND EXTRA_LIBS user32 gdi32 winmm imm32 ole32 oleaut32 version uuid advapi32 shell32)
-+  list(APPEND EXTRA_LIBS user32 gdi32 winmm imm32 ole32 oleaut32 version uuid advapi32 shell32 vcruntime)
- 
+   list(APPEND EXTRA_LIBS user32 gdi32 winmm imm32 ole32 oleaut32 version uuid advapi32 shell32)
+
++  if(MSVC AND NOT ${MSVC_VERSION} LESS 1920)
++    list(APPEND EXTRA_LIBS vcruntime)
++  endif()
++
    # TODO: in configure.in the check for timers is set on
    # cygwin | mingw32* - does this include mingw32CE?
+   if(SDL_TIMERS)

